storage/myisam/mi_open.c

```diff
@@ -1205,7 +1205,8 @@ uint mi_uniquedef_write(File file, MI_UNIQUEDEF *def)
   uchar buff[MI_UNIQUEDEF_SIZE];
   uchar *ptr=buff;
 
-  mi_int2store(ptr,def->keysegs);		ptr+=2;
+  mi_int2store(ptr,def->keysegs);		   ptr+=2;
+  mi_int2store(ptr,def->sql_key_no);   ptr+=2;
   *ptr++=  (uchar) def->key;
```

Comment:
I had asked Sergei about this. He said it will be okay if I add a new member to mi_uniquedef